### PR TITLE
Add option to automatically send replays in config

### PR DESCRIPTION
**Pull request by ascriptmaster**
Code ostensibly works, but I'm worried about race conditions and non-default clients. Is that something we need to worry about?

### CODE REVIEW

File: config/config-example.js [hunk 1]
```diff
@@ -96,6 +96,14 @@ exports.pmmodchat = false;
 //   This setting can also be turned on with the command /forcetimer.
 exports.forcetimer = false;
 
+// auto save replays - automatically attempt to send replays after a battle ends
+//   Useful for tournament servers so that users don't have to waste time saving
+//   the battles manually. Server must be registered.
+//
+// Your server *must* be registered in order to save replays, so don't turn this
+// on unless it is registered!
```
Comment (Zarel) [inline config/config-example.js]: Maybe add a bunch of warnings here,

or just leave it out of config-example completely and make it clear it's undocumented and not designed for public consumption

Comment (Zarel) [inline config/config-example.js]: I really do not want any server other than smogtours making use of this feature

**Comment by codelegend** [comment]
We can probably have more options, like `true` saves all replays, `'tours'` saves only for tournament battles, `'normal'` saves only for non-tour battles., `'rated'` for rated battles only etc.

File: config/config-example.js (hunk 1 above)
Comment (ascriptmaster) [inline config/config-example.js]: Reverted the file.

**Pull request merged**